com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveMainClassHandler.java

```diff
@@ -149,6 +161,66 @@ public void acceptSearchMatch(SearchMatch match) {
         return resolutions;
     }
 
+    private List<ResolutionItem> resolveMainClassUnderProject(final String projectName) {
+        // Limit to search main method from source code only.
+        IJavaSearchScope searchScope = SearchEngine.createJavaSearchScope(ProjectUtils.getJavaProjects(),
```

Comment:
Since here we now the `projectName` already, what about we find out the `IJavaProject` according to the `projectName`, and pass it as the first param of `SearchEngine.createJavaSearchScope()`?

Reply:
Good idea.